packages/react/src/components/Copy/Copy.js

```diff
@@ -6,91 +6,101 @@
  */
 
 import PropTypes from 'prop-types';
-import React, { Component } from 'react';
+import React, { useState, useEffect, useCallback } from 'react';
+import debounce from 'lodash/debounce';
 import classnames from 'classnames';
 import { settings } from 'carbon-components';
 
 const { prefix } = settings;
 
-export default class Copy extends Component {
-  static propTypes = {
-    /**
-     * Pass in content to be rendred in the underlying <button>
-     */
-    children: PropTypes.node,
-
-    /**
-     * Specify an optional className to be applied to the underlying <button>
-     */
-    className: PropTypes.string,
-
-    /**
-     * Specify the string that is displayed when the button is clicked and the
-     * content is copied
-     */
-    feedback: PropTypes.string,
-
-    /**
-     * Specify the time it takes for the feedback message to timeout
-     */
-    feedbackTimeout: PropTypes.number,
-
-    /**
-     * Specify an optional `onClick` handler that is called when the underlying
-     * <button> is clicked
-     */
-    onClick: PropTypes.func,
-  };
-
-  static defaultProps = {
-    feedback: 'Copied!',
-    feedbackTimeout: 2000,
-    onClick: () => {},
+export default function Copy({
+  children,
+  className,
+  feedback,
+  feedbackTimeout,
+  onClick,
+  ...other
+}) {
+  const [animation, setAnimation] = useState('');
+  const classNames = classnames(className, {
+    [`${prefix}--copy-btn--animating`]: animation,
+    [`${prefix}--copy-btn--${animation}`]: animation,
+  });
+  const handleFadeOut = useCallback(
+    debounce(() => {
+      setAnimation('fade-out');
+    }, feedbackTimeout),
+    [feedbackTimeout]
+  );
+  const handleClick = useCallback(
+    event => {
+      setAnimation('fade-in');
+      onClick(event);
+      handleFadeOut();
+    },
+    [onClick, handleFadeOut]
+  );
+  const handleAnimationEnd = event => {
```

Comment:
Style-nit: use function declarations for event handlers

Reply:
just wondering is there a reason behind this? a function declaration would be hoisted, so in that case we might as well be advocating for all function declarations to be made at the top of their scopes

Reply:
I think at this point it's just as a style consistency point tbh, if we have hard numbers one way or another could definitely update the style guide!

Reply:
if it isn't part of the style guide yet I think there should be a discussion about that rule. I lean more towards function expressions in these cases to avoid function hoisting

Reply:
@emyarod it is in the style guide over in: https://github.com/carbon-design-system/carbon/blob/master/docs/style.md#writing-a-component

Feel free to bring up reasons against it in our Slack channel or a PR! We only use this style since it's what React is currently using in their official documentation and there weren't any glaring reasons to stray from it.

Reply:
I don't see anything about avoiding function expressions for event handlers in the style guide or the React documentation. can you elaborate on that?